conan/conan.go

```diff
@@ -0,0 +1,127 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package conan
+
+import (
+    "github.com/package-url/packageurl-go"
+    "github.com/sirupsen/logrus"
+    "os"
+    "path/filepath"
+    "strings"
+)
+
+var (
+    newline = "\n"
+)
+
+type conanPurlInfo struct {
+    name string
+    version string
+}
+
+type Options struct {
+    Directory  string
+    BinaryName string
+}
+
+type ConanGenerator struct {
+    filepath string
+    logger *logrus.Logger
+}
+
+func New(logger *logrus.Logger, options Options) *ConanGenerator {
+    if options.Directory == "" {
+        options.Directory = "."
+    }
+
+    return &ConanGenerator{
+        filepath: filepath.Join(options.Directory, "conanfile." + options.BinaryName + ".cheque"),
+        logger: logger,
+    }
+}
+
+func (c ConanGenerator) CheckOrCreateConanFile(purls []packageurl.PackageURL) (e error) {
+    var err error
+    if _, err = os.Stat(c.filepath); os.IsNotExist(err) {
+        duplessPurls := c.checkForDuplicates(purls)
+        c.writeConanFile(duplessPurls)
+        err = nil
+    }
+    return err
+}
+
+func (c ConanGenerator) checkForDuplicates(purls []packageurl.PackageURL) []conanPurlInfo {
+    duplessMap := make(map[conanPurlInfo]bool)
+
+    //Add our lib prefix libraries
+    for _, purl := range purls[:] {
+        info := conanPurlInfo{
+            name:    purl.Name,
+            version: purl.Version,
+        }
+        if strings.HasPrefix(purl.Name, "lib") {
+            duplessMap[info] = true
+        }
+    }
+
+    //Check to see if our libless prefix's exist already in lib form.
+    for _, purl := range purls[:] {
+        info := conanPurlInfo{
+            name:    "lib" + purl.Name,
+            version: purl.Version,
+        }
+        if !strings.HasPrefix(purl.Name, "lib") && !duplessMap[info] {
+            duplessMap[info] = true
+        }
+    }
+
+    keys := make([]conanPurlInfo, 0, len(duplessMap))
+    for k := range duplessMap {
+        keys = append(keys, k)
+    }
+
+    return keys
+}
+
+func (c ConanGenerator) writeConanFile(purls []conanPurlInfo) {
+    file, err := os.Create(c.filepath)
+    if err != nil {
+        c.logger.Error(err)
+        return
+    }
+    defer file.Close()
+
+    header := "[requires]" + newline
+    n, err := file.WriteString(header)
+    if err != nil {
+        c.logger.Error(err)
+        return
+    }
+    if n != len(header) {
+        c.logger.Error("Unable to write data")
+        return
+    }
+    for _, purl := range purls {
+        purlInfo := purl.name + "/" + purl.version + newline
```

Comment:
Rather than create a new string you could do `fmt.Sprintf("%s/%s\n", purl.name, purl.version)`

Also to make it cross platform:

```
lineBreak := "\n"

if runtime.GOOS == "windows" {
  lineBreak = "\r\n"
}

fmt.Sprintf("%s/%s%s", purl.name, purl.version, lineBreak)
```

Reply:
done!

in java string format is very slow and not really recommended, so that fought against my instincts

Reply:
The fastest way is definitely the way you did it, but you are talking about 150 ns difference, which isn't enough to even care about here (which makes me pointing out streaming a total LOL):

https://dev.to/pmalhaire/concatenate-strings-in-golang-a-quick-benchmark-4ahh

In general, I find the `fmt.Sprintf()` style easier to read, to some degree, and also easier to adjust without fat fingering.